datafusion/physical-expr/src/planner.rs

```diff
@@ -339,3 +312,48 @@ pub fn create_physical_expr(
         ))),
     }
 }
+
+type InListCastResult = (Arc<dyn PhysicalExpr>, Vec<Arc<dyn PhysicalExpr>>);
+
+pub(crate) fn in_list_cast(
+    expr: Arc<dyn PhysicalExpr>,
+    list: Vec<Arc<dyn PhysicalExpr>>,
+    input_schema: &Schema,
+) -> Result<InListCastResult> {
+    let expr_type = &expr.data_type(input_schema)?;
+    let list_types: Vec<DataType> = list
+        .iter()
+        .map(|list_expr| list_expr.data_type(input_schema).unwrap())
+        .collect();
+    let result_type = get_coerce_type(expr_type, &list_types);
+    match result_type {
+        None => Err(DataFusionError::Internal(format!(
```

Comment:
```suggestion
        None => Err(DataFusionError::Plan(format!(
```

It seems like this could be a normal error (due to the query, for example) rather than an internal error